### PR TITLE
feat: [EXT-2278] add warning notice page for local development

### PR DESCRIPTION
**Pull request by Jwhiles**
Adding a 'landing screen' 
image

### CODE REVIEW

File: template/public/index.html [hunk 1]
```diff
@@ -15,11 +15,5 @@
       To begin the development, run `npm start`.
       To create a production bundle, use `npm run build`.
     -->
-    <script>
```
Comment (kdamball) [inline template/public/index.html:18]: I missed this - why are we removing the redirect?

File: template/src/index.tsx [hunk 2]
```diff
@@ -23,47 +23,55 @@ import Sidebar from './components/Sidebar';
 import Field from './components/Field';
 import Dialog from './components/Dialog';
 
-init((sdk: BaseExtensionSDK) => {
+import LocalhostWarning from './components/LocalhostWarning';
+
+if (process.env.NODE_ENV === 'development' && window.self === window.top) {
+  // You can remove this if block before deploying your app
   const root = document.getElementById('root');
```
Comment (kdamball) [inline template/src/index.tsx:30]: fwiw - this can go outside the if statement.

File: template/public/index.html (hunk 1 above)
Comment (Jwhiles) [reply template/public/index.html:18]: In order to show the explanation warning screen :)

Comment (stefanjudis) [reply template/public/index.html:18]: We've seen several people in the community being confused by the redirect. First, they don't understand that they shouldn't open `localhost` directly and if they do, they're completely lost with the target of the redirect. :)

File: template/src/components/LocalhostWarning.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <div
+      style={{
+        display: 'flex',
+        justifyContent: 'center',
+        marginTop: '40px'
+      }}>
+      <Note title="It looks like you're running your app locally." style={{ maxWidth: '800px' }}>
+        <Paragraph>
+          If this is your first time running your app, please run `npx
+          @contentful/create-contentful-app create-defintion` in your terminal then{' '}
```
Comment (davidfateh) [inline template/src/components/LocalhostWarning.tsx:15]: While I think this is a good approach, I still feel a bit hesitant about this appDefinition flow. I think for first-timers, this is still pretty confusing and the appDefinition might make more sense when they see the UI version of it.

That being said, we can always gather more feedback once this version is released to see if people are still confused by it.

**Review by davidfateh — APPROVED** [review]

File: template/src/components/LocalhostWarning.tsx (hunk 3 above)
Comment (stefanjudis) [reply template/src/components/LocalhostWarning.tsx:15]: ☝️ I agree with this. I'd favor to give two options here:

- a link to documentation explaining the concept and the steps in the ui
- a "alright, you know what ya doing" way promoting the cli command

File: template/src/index.tsx [hunk 4]
```diff
@@ -23,47 +23,55 @@ import Sidebar from './components/Sidebar';
 import Field from './components/Field';
 import Dialog from './components/Dialog';
 
-init((sdk: BaseExtensionSDK) => {
+import LocalhostWarning from './components/LocalhostWarning';
+
+if (process.env.NODE_ENV === 'development' && window.self === window.top) {
```
Comment (kdamball) [inline template/src/index.tsx:28]: Do we care if we're not in Dev mode? i.e: before we'd redirect even when not in dev mode

Comment (kdamball) [reply template/src/index.tsx:28]: Otherwise - it looks good to me

File: template/src/index.tsx [hunk 5]
```diff
@@ -23,47 +23,55 @@ import Sidebar from './components/Sidebar';
...
-  // doesn't require it
-  ComponentLocationSettings.forEach(componentLocationSetting => {
-    if (sdk.location.is(componentLocationSetting.location)) {
-      render(componentLocationSetting.component, root);
-    }
+    // Select a component depending on a location in which the app is rendered.
+    //
+    // NB: Location "app-config" is auto-included in the list as most apps need it
+    // You can remove it (and on the app definition also) in case the app
+    // doesn't require it
```
Comment (VaguelySerious) [inline template/src/index.tsx:70]: How is it auto included? It seems to be just an element in the array like any other location

**Pull request merged**